beater/beater.go

```diff
@@ -618,12 +620,26 @@ func newTransformConfig(beatInfo beat.Info, cfg *config.Config) (*transform.Conf
 }
 
 func newSourcemapStore(beatInfo beat.Info, cfg config.SourceMapping) (*sourcemap.Store, error) {
-	esClient, err := elasticsearch.NewClient(cfg.ESConfig)
+	if len(cfg.SourceMapConfigs) == 0 {
```

Comment:
Doesn't that lead to a similar problem we faced with the agent config settings? If the sourcemap gets uploaded after the APM Server was started, APM Server would still try to fetch from the ES store instead of the fleet store.

Reply:
I'm not sure. If a sourcemap is uploaded, then the config will be regenerated and pushed to apm-server, right? That will trigger a reload of the server and fresh evaluation of this logic, which should now have the sourcemap config.

Reply:
That's right. But when the APM Server is in fleet mode and no sourcemaps are configured, the server would still query ES to see if a sourcemap is there, when RUM events are ingested. In the fleet mode, the server should never run queries against ES. Instead, if no sourcemaps are configured, the fetcher simply shouldn't query anything, as there is no entry for the key.

Reply:
Cool, sounds good